app/src/main/java/ch/epfl/sweng/favors/EditProfileFragment.java

```diff
@@ -73,17 +77,19 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
 
          binding.profCityEdit.addTextChangedListener(profCityEditWatcher);
 
-         binding.profSexEdit.addTextChangedListener(profSexEditWatcher);
+         binding.profGenderEdit.setOnCheckedChangeListener((RadioGroup group, int checkedId) ->{
+                    if(checkedId == R.id.profGenderMEdit){
+                        User.UserGender.setGender(User.getMain(),User.UserGender.M);
+                    }
+                    if(checkedId == R.id.profGenderFEdit){
```

Comment:
why not making it an else if here and avoid the additional unnecessary check?
Or even better since you made it an enum to abstract from the database implementation (which as bespoken I think is a very good move and practice) match on it in a case statement ...